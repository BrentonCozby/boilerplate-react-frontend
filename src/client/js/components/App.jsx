import React, { Component } from 'react'
import { Route, Switch } from 'react-router-dom'
import { rootUrl } from '../../../../config.js'

import Menu from '../containers/Menu.jsx'
import NoMatch from './NoMatch.jsx'
import Footer from './Footer.jsx'
import Home from '../containers/Home.jsx'

class App extends Component {
    render() {
        return (
            <div className="App">
                <Menu />
                <Switch>
                    <Route exact path={`${rootUrl}/`} component={Home} />
                    <Route component={NoMatch}/>
                </Switch>
                <Footer />
            </div>
        )
    }
}

export default App
