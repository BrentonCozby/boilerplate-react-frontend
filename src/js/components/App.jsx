import React, { Component } from 'react'
import { Route, Switch } from 'react-router-dom'
import { PP } from '../../../config.js'

import Menu from './Menu/index.jsx'
import NoMatch from './NoMatch.jsx'
import Footer from './Footer.jsx'
import Home from './Home/index.jsx'

class App extends Component {
    render() {
        return (
            <div className="App">
                <Menu />
                <Switch>
                    <Route exact path={`${PP}/`} component={Home} />
                    <Route component={NoMatch}/>
                </Switch>
                <Footer />
            </div>
        )
    }
}

export default App
