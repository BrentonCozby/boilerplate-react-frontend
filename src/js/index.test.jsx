import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { BrowserRouter as Router, Route } from 'react-router-dom'

import configStore from './store.js'
import App from './components/app-component.jsx'

describe('app-component', () => {
    it('renders App with router and store without crashing', () => {
        const store = configStore()

        const wrappedApp = renderer.create(
            <Provider store={store}>
                <Router>
                    <Route path={PP} component={App} />
                </Router>
            </Provider>
        )
        
        expect(wrappedApp.toJSON()).toMatchSnapshot()
    })
})