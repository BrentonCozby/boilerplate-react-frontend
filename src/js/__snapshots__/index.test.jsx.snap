// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app-component renders App with router and store without crashing 1`] = `
<div
  className="App"
>
  <div>
    <div
      className="menu-button"
      onClick={[Function]}
    >
      <span
        className="bar"
      />
      <span
        className="bar"
      />
      <span
        className="bar"
      />
    </div>
    <div
      className="Menu"
    >
      <a
        className="Menu-item"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </div>
  </div>
  <div
    className="NoMatch"
  >
    <h1>
      Page not found
    </h1>
  </div>
  <footer
    className="Footer"
  >
    <p
      className="copyright"
    >
      Â© 
      2017
      . Brenton Cozby. View more projects at 
      <a
        href="https://brentoncozby.com"
      >
        brentoncozby.com
      </a>
      .
    </p>
  </footer>
</div>
`;
